perllib/FixMyStreet/Cobrand/Kingston.pm

```diff
@@ -0,0 +1,608 @@
+        #535 => 6,
+        #536 => 4,
+        #537 => 6,
+        #541 => 4,
+        #542 => 6,
+        #544 => 4,
+        #545 => 6,
+    );
+
+    $self->{c}->stash->{quantity_max} = \%quantity_max;
+
+    $self->{c}->stash->{garden_subs} = $self->waste_subscription_types;
+
+    my $result = $self->{api_serviceunits};
+    return [] unless @$result;
+
+    my $events = $self->_parse_events($self->{api_events});
+    $self->{c}->stash->{open_service_requests} = $events->{enquiry};
+
+    # If there is an open Garden subscription (2106) event, assume
+    # that means a bin is being delivered and so a pending subscription
+    $self->{c}->stash->{pending_subscription} = $events->{enquiry}{2106} ? { title => 'Garden Subscription' } : undef;
+
+    my @to_fetch;
+    my %schedules;
+    my @task_refs;
+    my %expired;
+    foreach (@$result) {
+        my $servicetasks = $self->_get_service_tasks($_);
+        foreach my $task (@$servicetasks) {
+            my $service_id = $task->{TaskTypeId};
+            my $schedules = _parse_schedules($task);
+            $expired{$service_id} = $schedules if $self->waste_sub_overdue( $schedules->{end_date}, weeks => 4 );
+
+            next unless $schedules->{next} or $schedules->{last};
+            $schedules{$service_id} = $schedules;
+            push @to_fetch, GetEventsForObject => [ ServiceUnit => $_->{Id} ];
+            push @task_refs, $schedules->{last}{ref} if $schedules->{last};
+        }
+    }
+    push @to_fetch, GetTasks => \@task_refs if @task_refs;
+
+    my $cfg = $self->feature('echo');
+    my $echo = Integrations::Echo->new(%$cfg);
+    my $calls = $echo->call_api($self->{c}, 'kingston', 'bin_services_for_address:' . $property->{id}, @to_fetch);
+
+    my @out;
+    my %task_ref_to_row;
+    foreach (@$result) {
+        my $servicetasks = $self->_get_service_tasks($_);
+        foreach my $task (@$servicetasks) {
+            my $service_id = $task->{TaskTypeId};
+            my $service_name = $self->service_name_override($service_id);
+            next unless $schedules{$service_id} || ( $service_name eq 'Garden Waste' && $expired{$service_id} );
+
+            my $schedules = $schedules{$service_id} || $expired{$service_id};
+
+            my $containers = $service_to_containers{$service_id};
+            my ($open_request) = grep { $_ } map { $events->{request}->{$_} } @$containers;
+
+            my $request_max = $quantity_max{$service_id};
+
+            my $garden = 0;
+            my $garden_bins;
+            my $garden_cost = 0;
+            my $garden_due = $self->waste_sub_due($schedules->{end_date});
+            my $garden_overdue = $expired{$_->{Id}};
+            if ($service_name eq 'Garden Waste') {
+                $garden = 1;
+                my $data = Integrations::Echo::force_arrayref($task->{Data}, 'ExtensibleDatum');
+                foreach (@$data) {
+                    next unless $_->{DatatypeName} eq 'LBB - GW Container'; # DatatypeId 5093
```

Comment:
I guess we don't know what this should be yet?

Reply:
I think it's "SLWP - Containers" (it appears to be that in live but can't find same in staging).